test/test_sparse.py

```diff
@@ -4446,6 +4446,22 @@ def test_zeros_like_fake(self, dtype, layout):
                 result = torch.zeros_like(f, device=f.fake_device)
             self.assertEqual(result, expected)
 
+    @all_sparse_layouts('layout', include_strided=False)
+    @parametrize("dtype", [torch.float64])
+    def test_empty_like_fake(self, dtype, layout):
+        from torch._subclasses.fake_tensor import FakeTensorMode, FakeTensor
+        from torch.utils._mode_utils import no_dispatch
+        fake_mode = FakeTensorMode()
+        index_dtype = torch.int64
+        device = 'cpu'
+        for t in self.generate_simple_inputs(layout, device=device, dtype=dtype, index_dtype=index_dtype):
+            f = FakeTensor.from_tensor(t, fake_mode)
+            # empty_like on a fake sparse tensor is equivalent to zeros_like:
+            expected = torch.zeros_like(t)
+            with no_dispatch():
+                result = torch.empty_like(f, device=f.fake_device)
```

Comment:
Overall this looks okay to me, but I am curious if we have any precedent for an operator to have different behavior w/ `FakeTensor`.

Reply:
AFAICT, applying `empty_like` to a fake tensor with non-fake device is never exercised in the current code base.

The different behavior occurs only when using `empty_like` to create a real tensor from a fake tensor.

This behavioral difference could be eliminated by either making empty_like to return a zero tensor because fake tensor does not carry indices data other than their shapes, or by forcing fake tensor to carry also indices data (as a part of spars tensor shape data).

It is OT for this PR but as I mentioned in the source code comments, it feels like our past design decision for empty_like to return the indices of the input could have been wrong because it leads to a number of inconsistencies at various levels of using empty_like on sparse tensors.